src/intersections/triangles.jl

```diff
@@ -0,0 +1,127 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    intersecttype(s, t)
+
+Calculate the intersection of a segment and triangle in 3D.
+Algorithm from Jiménez et al 2009 (A robust segment/triangle intersection algorithm for interference tests. Efficiency study). 
```

Comment:
Please follow our citation style, here is an example: https://github.com/JuliaGeometry/Meshes.jl/blob/d2e312300eb8ab75c8d8baebe600aaf56004dd6b/src/intersections/segments.jl#L19-L23